Minesweeper.py

```diff
@@ -44,132 +35,133 @@ def __init__(self):
-        # DHB: initialising a library should not be done in an object initialiser like this. Move it to file scope, or move everything on file scope into this class (which would be a bad choice)
-        pygame.init()
-        # DHB: you're not really use the game_objects system... use it, or remove it  
         self.game_objects = []
-        # DHB: it is important to comment all member variables, explaining what they are used for and what they can contain, just like all functions and classes.
         self.tiles = None
         self.game_over = False
         self.game_won = False
-        self.game_won_message_printed = False
-        # DHB: for a state variable like this, it's especially important to ensure that the valid states are well defined. You could for example use a game state
-        # DHB: enumeration class, similar to the color class, to define the valid status
         self.state = "tutorial"
         self.clock = pygame.time.Clock()
         self.time_elapsed = 0
         self.start_time = 0
+        pygame.mouse.set_visible(True)
+        pygame.mouse.set_cursor(*pygame.cursors.tri_left)
 
         self.tutorial_screen = TutorialScreen()
         self.game_won_screen = GameWonScreen()
         self.reset_button = ResetButton()
 
     def add_game_objects(self, *list_of_game_objects, **dict_of_game_objects):
+        """ adds amount of game objects
+        """
         for game_object in list_of_game_objects:
             self.add_game_object(game_object)
         for game_object_id, game_object in dict_of_game_objects.items():
             self.add_game_object(game_object)
 
     def add_game_object(self, game_object):
+        """adds individual game object
+        """
         self.game_objects.append(game_object)
         if isinstance(game_object, Tiles):
             self.tiles = game_object
 
-    def get_game_objects_with_attribute(self, attribute_name):
-        return [game_object for game_object in self.game_objects if hasattr(game_object, attribute_name)]
-
-    def draw(self): # calls all of the draw functions
+    def draw(self):
+        """calls the draw functions required under certain circumstances
+        """
         
         if self.game_over:
             self.tiles.draw()
-
-        # DHB: I wonder if this 'game_won_message_printed' flag is a good idea. Draw loops usually draw everything
-        # DHB: in every frame, but I imagine this will only make the message be drawn in one frame 
-        if self.game_won and not self.game_won_message_printed:
+        
+        if self.game_won:
             self.game_won_screen.draw()
-            self.game_won_message_printed = True
-
         self.reset_button.draw()
 
-    def get_remaining_flags(self):
-        # DHB: it would be wise to name this count_remaining_flags, rather than get_remaining_flags, as get implies that the state already
-        # DHB: was calculated. This function is heavy for a 'get' method.
+        pygame.display.update
```

Comment:
you are just referring to the name of the update function here, not actually calling it. This code does nothing.